e2e-tests/tests/monitoringconfig_pg/01-assert.yaml

```diff
@@ -0,0 +1,34 @@
+apiVersion: kuttl.dev/v1
+kind: TestAssert
+timeout: 300
+---
+apiVersion: everest.percona.com/v1alpha1
+kind: DatabaseCluster
+metadata:
+  name: pg-mc
+spec:
+  monitoringConfigName: test-mc
+status:
+  status: ready
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: everest-secrets-pg-mc
+data:
+  pmmserverkey: YWJjZGVm
+---
+apiVersion: pgv2.percona.com/v2
+kind: PerconaPGCluster
+metadata:
+  name: pg-mc
+spec:
+  pmm:
+    enabled: true
+    serverHost: http://localhost
+    image: percona/pmm-client:latest
+    secret: everest-secrets-pg-mc
+    resources:
+      limits:
+        memory: "128Mi"
+        cpu: "500m"
```

Comment:
It misses asserts for the status field that a database cluster getting ready. Having a ready cluster will improve tests and other steps that ensure that it changes something when we change a secret or monitoring config should also assert that a database cluster changed its state to `initializing`.

Another solution to ensure that reconcile loop was triggered is to check the generation field in the metadata and ensure that it increments. In that case tests will check that it actually changes something in the databasecluster CR

Reply:
However, a database cluster may not change the state to `initializing` and generation may be not the best thing to run asserts against.
https://stackoverflow.com/a/66092577 you can choose the best option from this list